pyspark_datasources/sharepoint.py

```
@@ -0,0 +1,415 @@
+    - **Stream processing**: Uses Microsoft Graph API for efficient concurrent writes
+    - **Exactly-once semantics**: Integrates with Spark's checkpoint mechanism
+    - **Error handling**: Control over whether to fail the write operation if a record fails to be written
+    - **Flexible resource implementations**: Supports multiple resource types (currently only "list")
+    """
+
+    @classmethod
+    def name(cls) -> str:
+        """Return the short name for this Sharepoint datasource."""
+        return "pyspark.datasource.sharepoint"
+
+    def streamWriter(self, schema: StructType, overwrite: bool) -> "SharepointStreamWriter":
```

Ruff (0.14.1)

Unused method argument: `schema`

(ARG002)